### ISSUE TITLE
False-positive `invalid-type-form` errors for type aliases inside `Literal` slices

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ty#1368]
### Summary

This issue reproduces on `main` if you use PEP-695 type aliases, and starts causing a lot more false positives on https://github.com/astral-sh/ruff/pull/20107. Pyright and mypy are both fine with this snippet, but we currently emit `invalid-type-form` on it:

```py
from typing import Literal
type X = Literal["foo"]

# error: [invalid-type-form] "Type arguments for `Literal` must be `None`, a literal value (int, bool, str, or bytes), or an enum member"
x: Literal[X]
```
